### PR TITLE
Home work 13

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import java.util.*;
```
Comment (Sofasmile) [inline src/Main.java]: better doesn't use autoimport

File: src/Main.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+public class Main {
+
+	public static void main(String[] args) throws Exception {
```
Comment (Sofasmile) [inline src/Main.java]: try to simplify this method

File: src/Main.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		StringBuilder builder = new StringBuilder();
+		Scanner scanner = new Scanner(System.in);
+		while (scanner.hasNextLine()) {
```
Comment (Sofasmile) [inline src/Main.java]: it is unnecessary to use 'while' cycle

File: src/Main.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		for (Map.Entry<String, Integer> entry : wordsByNumber.entrySet()) {
+			String word = entry.getKey();
+			Integer number = entry.getValue();
+			if (!numberByWords.containsKey(number)) {
+				numberByWords.put(number, new ArrayList<>());
+			}
+			numberByWords.get(number).add(word);
+		}
+		List<String> allWords = new ArrayList<>();
+		numberByWords.keySet().stream()
```
Comment (Sofasmile) [inline src/Main.java]: read about method comparingByValue() and comparingByKey()
try to use it for sorting

Comment (UnitOmega13) [reply src/Main.java]: I have changed all method

File: src/Main.java (hunk 1 above)
Comment (Sofasmile) [reply src/Main.java]: What about it?

File: src/Main.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Main {
+
+	public static void main(String[] args) throws Exception {
```
Comment (Sofasmile) [inline src/Main.java]: do we need to throws exception here?

Comment (UnitOmega13) [reply src/Main.java]: fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**